lib/metasploit/concern/loader.rb

```diff
@@ -109,19 +102,25 @@ def register
   # @param pathname [Pathname] a Pathname under {#root}.
   # @return [Object] if {#pathname_to_constant_name} returns a constant name
   # @return [nil] otherwise
-  def constantize_pathname(mechanism:, pathname:)
+  def constantize_pathname(pathname:)
     constant_name = pathname_to_constant_name(pathname)
 
     constant = nil
 
     if constant_name
-      # require before calling constantize so that the constant isn't recorded as unloadable.
-      if mechanism == :require
-        require pathname
+      begin
+        # constantize either way as the the constant_name still needs to be converted to Module
+        constant = constant_name.constantize
+      rescue => e # rescue any here should probably be more specific
+        if e.kind_of?(NameError)
+          begin
+            require pathname
+            constant = constant_name.constantize
+          rescue NameError => _e
+            _e # left for debugging breakpoint
```

Comment:
```suggestion
            _e # left for debugging breakpoint
            nil
```

Reply:
This will always fall thru to exit the `if` and `constant` is returned at the end of the method.